testdata/sqllogictests/cast/oid.slt

```diff
@@ -0,0 +1,30 @@
+
+statement error
+select 'my_table'::oid;
+
+statement ok
+create external table my_table from debug options(table_type = 'never_ending');
+
+statement ok
+select 'my_table'::oid;
+
+# make sure upper case works
+statement ok
+select 'my_table'::OID;
+
+statement ok
+select cast('my_table' as oid);
+
+statement ok
+select cast('my_table' as OID);
+
+
+# We don't support expressions other than strings right now.
```

Comment:
is there a point to supporting more ever/eventually?

Reply:
idk, most of this was copied from the `regclass` stuff, so i didn't want to change any more than absolutely necessary to implement the new functionality